src/Workspaces/Core/Portable/SymbolKey/SymbolKey.FunctionPointerTypeSymbolKey.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial struct SymbolKey
+    {
+        private static class FunctionPointerTypeSymbolKey
+        {
+            public static void Create(IFunctionPointerTypeSymbol symbol, SymbolKeyWriter visitor)
+            {
+                visitor.WriteRefKind(symbol.Signature.RefKind);
+                visitor.WriteSymbolKey(symbol.Signature.ReturnType);
+                visitor.WriteRefKindArray(symbol.Signature.Parameters);
+                visitor.WriteParameterTypesArray(symbol.Signature.Parameters);
+            }
+
+            public static SymbolKeyResolution Resolve(SymbolKeyReader reader)
+            {
+                var returnRefKind = reader.ReadRefKind();
+                var returnType = reader.ReadSymbolKey();
+                using var paramRefKinds = reader.ReadRefKindArray();
+                var paramTypes = reader.ReadSymbolKeyArray<ITypeSymbol>();
+
+                if (paramTypes.IsDefault)
+                {
+                    return default;
+                }
+
+                using var _ = paramTypes;
```

Comment:
should this not be above the .IsDefault check?

Reply:
It's explicitly below, because if `paramTypes` is default there's nothing to dipose.

Reply:
docs say:

```
            /// Callers should <see cref="IDisposable.Dispose"/> the instance returned.  No check is
            /// necessary if <c>default</c> was returned before calling <see cref="IDisposable.Dispose"/>
            /// </summary>
```

Reply:
and all existing callers put this in a `using`.  so i'd far prefer you keep your code using that pattern and keeping things simple that way.  thanks!

Reply:
Done.